File: src/main/java/com/youtube/tracker/models/YoutubeSearchResponse.java
package com.youtube.tracker.models;

import lombok.Value;

import java.time.OffsetDateTime;
import java.util.List;

@Value
public class YoutubeSearchResponse {
    String kind;
    String etag;
    String nextPageToken;
    String regionCode;
    PageInfo pageInfo;
    List<Item> items;

    @Value
    public static class Item {
        String kind;
        String etag;
        Id id;
        Snippet snippet;
    }

    @Value
    public static class Id {
        String kind;
        String videoId;
    }

    @Value
    public static class Snippet {
        OffsetDateTime publishedAt;
        String channelId;
        String title;
        String description;
        Thumbnails thumbnails;
        String channelTitle;
        String liveBroadcastContent;
        OffsetDateTime publishTime;
    }

    @Value
    public static class Thumbnails {
        Default thumbnailsDefault;
        Default medium;
        Default high;
    }

    @Value
    public static class Default {
        String url;
        long width;
        long height;
    }

    @Value
    public static class PageInfo {
        long totalResults;
        long resultsPerPage;
    }
}

File: src/main/kotlin/com/youtube/tracker/models/response/YoutubeSearchResponse.kt
package com.youtube.tracker.models.response

import java.time.OffsetDateTime

data class YoutubeSearchResponse(
    val kind: String? = null,
    val etag: String? = null,
    val nextPageToken: String? = null,
    val regionCode: String? = null,
    val pageInfo: PageInfo? = null,
    val items: List<Item>? = null
)

data class Item(
    val kind: String? = null,
    val etag: String? = null,
    val id: Id? = null,
    val snippet: Snippet? = null
)

data class Id(
    val kind: String? = null,
    val videoId: String? = null
)

data class Snippet(
    val publishedAt: OffsetDateTime? = null,
    val channelId: String? = null,
    val title: String? = null,
    val description: String? = null,
    val thumbnails: Thumbnails? = null,
    val channelTitle: String? = null,
    val liveBroadcastContent: String? = null,
    val publishTime: OffsetDateTime? = null
)

data class Thumbnails(
    val thumbnailsDefault: Default? = null,
    val medium: Default? = null,
    val high: Default? = null
)

data class Default(
    val url: String? = null,
    val width: Long = 0,
    val height: Long = 0,
)

data class PageInfo(
    val totalResults: Long = 0,
    val resultsPerPage: Long = 0
)